### PR TITLE
Fix request_processor and add tests for it.

### PR DESCRIPTION
**Pull request by siarheishut**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -32,7 +32,7 @@ add_executable(route_manager_tests
         src/bus_manager.cpp
         tests/request_parser_test.cpp
         tests/bus_manager_test.cpp
-        tests/test_utils.cpp)
+        tests/test_utils.cpp tests/request_processor_test.cpp)
```
Comment (foxfrikses) [inline CMakeLists.txt:35]: \n

File: src/request_processor.cpp [hunk 2]
```diff
@@ -4,55 +4,54 @@
 #include <variant>
 
 namespace rm {
-json::Dict ProcessBusRequest(const BusManager &bm,
-                             const GetBusRequest &request) {
+json::Dict ToJson(std::optional<BusResponse> resp, int id) {
   json::Dict response;
```
Comment (foxfrikses) [inline src/request_processor.cpp:8]: Let's rename them: `json::Dict response`, `std::optional<BusResponse> response`

File: src/request_processor.cpp [hunk 3]
```diff
@@ -4,55 +4,54 @@
...
+  response.emplace("route_length", resp->length);
+  response.emplace("curvature", resp->curvature);
+  response.emplace("stop_count", resp->stop_count);
+  response.emplace("unique_stop_count", resp->unique_stop_count);
   return response;
 }
 
-json::Dict ProcessStopRequest(const BusManager &bm,
-                              const GetStopRequest &request) {
+json::Dict ToJson(std::optional<StopResponse> resp, int id) {
```
Comment (foxfrikses) [inline src/request_processor.cpp:22]: same renaming as above

File: tests/request_processor_test.cpp [hunk 4]
```diff
@@ -0,0 +1,88 @@
+#include <gtest/gtest.h>
+
+#include "src/request_processor.h"
+
+#include <string>
+#include <vector>
+#include <optional>
+
+TEST(TestProcessRequests, TestStopRequest) {
+  using Response = std::optional<rm::StopResponse>;
```
Comment (foxfrikses) [inline tests/request_processor_test.cpp:10]: `ResponseOpt`

File: tests/request_processor_test.cpp [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+  };
+
+  for (auto &[name, response, id, want] : test_cases) {
+    json::Dict got = rm::ToJson(response, id);
+    EXPECT_EQ(want, got);
+  }
+}
+
+TEST(TestProcessRequests, TestBusRequest) {
+  using Response = std::optional<rm::BusResponse>;
```
Comment (foxfrikses) [inline tests/request_processor_test.cpp:50]: `ResponseOpt`

**Review by foxfrikses — CHANGES_REQUESTED** [review]

File: CMakeLists.txt (hunk 1 above)
Comment (siarheishut) [reply CMakeLists.txt:35]: Done.

File: src/request_processor.cpp (hunk 2 above)
Comment (siarheishut) [reply src/request_processor.cpp:8]: Done.

File: src/request_processor.cpp (hunk 3 above)
Comment (siarheishut) [reply src/request_processor.cpp:22]: Done.

File: tests/request_processor_test.cpp (hunk 4 above)
Comment (siarheishut) [reply tests/request_processor_test.cpp]: Done.

File: tests/request_processor_test.cpp (hunk 5 above)
Comment (siarheishut) [reply tests/request_processor_test.cpp]: Done.

File: tests/request_processor_test.cpp [hunk 6]
```diff
@@ -0,0 +1,88 @@
+#include <gtest/gtest.h>
+
+#include "src/request_processor.h"
+
+#include <string>
+#include <vector>
+#include <optional>
+
+TEST(TestProcessRequests, TestStopRequest) {
```
Comment (foxfrikses) [inline tests/request_processor_test.cpp:9]:
```suggestion
TEST(TestProcessRequests, TestStopResponseToJson) {
```

Comment (foxfrikses) [inline tests/request_processor_test.cpp:9]:
```suggestion
TEST(TestProcessRequests, TestStopResponseToJson) {
```

File: tests/request_processor_test.cpp [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+      },
+  };
+
+  for (auto &[name, response, id, want] : test_cases) {
+    json::Dict got = rm::ToJson(response, id);
+    EXPECT_EQ(want, got);
+  }
+}
+
+TEST(TestProcessRequests, TestBusRequest) {
```
Comment (foxfrikses) [inline tests/request_processor_test.cpp:49]:
```suggestion
TEST(TestProcessRequests, TestBusResponseToJson) {
```

**Review by foxfrikses — CHANGES_REQUESTED** [review]

**Review by foxfrikses — APPROVED** [review]
LGTM

**Pull request merged**